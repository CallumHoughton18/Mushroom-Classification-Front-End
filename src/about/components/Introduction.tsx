import React, {FunctionComponent, Fragment} from "react";
import Container from "../../shared/components/Layout/Container";
import Mushroom from "../../shared/components/Icons/Mushroom";

type introductionPropsType = {
    mushroomColor: string;
};
const Introduction: FunctionComponent<introductionPropsType> = ({mushroomColor}) => {
    const IntroductionText = (
        <Fragment>
            <h1>Mushroom Classification - A Machine Learning Example</h1>
            <p>
                This application is the front end to interact with the{" "}
                <a href="https://github.com/CallumHoughton18/Mushroom-Classification">
                    mushroom classification RESTful API
                </a>{" "}
                Which uses a model exposed via Flask, a web framework for Python, to predict whether
                a mushroom is poisonous.
                <b> It is NOT meant to be USED to determine if a mushroom is edible or not!</b> It
                is instead meant to demonstrate how a machine learning model can be exposed and
                interacted with. The application owner is not responsible for your use of the
                application.
            </p>
        </Fragment>
    );

    const introIcon = <Mushroom color={mushroomColor} className="mushroom" />;

    return (
        <Container
            sectionStyle="inner-content"
            InnerContent={IntroductionText}
            OuterContent={introIcon}
        ></Container>
    );
};

export default Introduction;
