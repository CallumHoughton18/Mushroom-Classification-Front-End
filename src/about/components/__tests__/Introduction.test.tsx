import React from "react";
import {render} from "@testing-library/react";

import Introduction from "../Introduction";

describe("<Introduction /> render test", () => {
    test("should display introduction text and graphic", () => {
        const renderedIntroduction = render(<Introduction />);
        expect(renderedIntroduction).toMatchSnapshot();
    });
});
