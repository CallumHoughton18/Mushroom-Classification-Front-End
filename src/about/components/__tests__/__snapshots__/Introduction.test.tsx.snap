// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Introduction /> render test should display introduction text and graphic 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="inner-content"
      >
        <div>
          <h1>
            Mushroom Classification - A Machine Learning Example
          </h1>
          <p>
            This application is the front end to interact with the
             
            <a
              href="https://github.com/CallumHoughton18/Mushroom-Classification"
            >
              mushroom classification RESTful API
            </a>
             
            Which uses a model exposed via Flask, a web framework for Python, to predict whether a mushroom is poisonous.
            <b>
               It is NOT meant to be USED to determine if a mushroom is edible or not!
            </b>
             It is instead meant to demonstrate how a machine learning model can be exposed and interacted with. The application owner is not responsible for your use of the application.
          </p>
        </div>
        <i
          class="fa"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="inner-content"
    >
      <div>
        <h1>
          Mushroom Classification - A Machine Learning Example
        </h1>
        <p>
          This application is the front end to interact with the
           
          <a
            href="https://github.com/CallumHoughton18/Mushroom-Classification"
          >
            mushroom classification RESTful API
          </a>
           
          Which uses a model exposed via Flask, a web framework for Python, to predict whether a mushroom is poisonous.
          <b>
             It is NOT meant to be USED to determine if a mushroom is edible or not!
          </b>
           It is instead meant to demonstrate how a machine learning model can be exposed and interacted with. The application owner is not responsible for your use of the application.
        </p>
      </div>
      <i
        class="fa"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
